core/store/src/flat/types.rs

```diff
@@ -25,9 +25,13 @@ impl BlockInfo {
 
 #[derive(strum::AsRefStr, Debug, PartialEq, Eq)]
 pub enum FlatStorageError {
-    /// This means we can't find a path from `flat_head` to the block. Includes `flat_head` hash and block hash,
-    /// respectively.
+    /// This means we can't find a path from `flat_head` to the block. Includes
+    /// `flat_head` hash and block hash, respectively.
+    /// Should not result in node panic, because flat head can move during processing
+    /// of some chunk.
```

Comment:
Hm, is my understanding correct that we should only use this error in this specific case where flat head moved while we were processing a fork? Because `BlockNotSupported` is a very generic name that can easily be used in other contexts, if someone is not aware of this and just looks at the name.

Would it perhaps make sense to rename the error to describe the situation more precisely? Such that nobody will accidentally use this error in other contexts?

Reply:
From `FlatStorage` API perspective we don't know if the block is not longer supported because flat head has moved or if it is just a wrong block which was never supported. So I think `BlockNotSupported` is pretty precise.